test/object-store/util/sync/redirect_server.hpp

```diff
@@ -60,9 +84,59 @@ class RedirectingHttpServer {
         m_server_thread.join();
     }
 
+    void set_event_hook(std::function<void(Event, std::optional<std::string>)> hook)
+    {
+        m_hook = hook;
+    }
+
+    // If true, http (app services) requests will first hit the redirect server and
+    // receive a redirect response which will contain the location to the actual
+    // server.
+    // NOTE: some http transport redirect implementations may strip the authorization
+    // header from the request after it is redirected and the user will be logged out
+    // from the client app as a result.
+    void force_http_redirect(bool remote = true)
+    {
+        m_http_redirect = remote;
+    }
+
+    // If true, websockets will be first directed to the redirect server which will
+    // return a redirect close code. The client will then update the location by
+    // querying the actual server location endpoint (from the 'hostname' location
+    // value) and open a websocket conneciton to the actual server.
+    // NOTE: the websocket will never connect if both http and websockets are
+    // redirecting and will just keep getting the redirect close code.
+    void force_websocket_redirect(bool force = true)
+    {
+        m_websocket_redirect = force;
+    }
+
+    // Reset the state to default (initial) state after test makes updates
+    void reset_state()
+    {
+        m_http_redirect = false;
+        m_websocket_redirect = false;
+        m_hook = nullptr;
+    }
+
     std::string base_url() const
     {
-        return util::format("http://localhost:%1", m_endpoint.port());
+        return m_base_url;
+    }
+
+    std::string server_url() const
+    {
+        return m_redirect_to_base_url;
+    }
+
+    std::string location_hostname() const
```

Comment:
this seems like something a caller of this type should already know?

Reply:
This returns the values that the redirect server `/location` endpoint will use so it can be displayed when using the redirect server command line tool - removed.